Model/src/main/java/org/gusdb/wdk/model/user/dataset/event/repo/UserDatasetEventRepo.java

```diff
@@ -0,0 +1,440 @@
+import org.gusdb.wdk.model.user.dataset.event.model.EventRow;
+
+/**
+ * Repository for DB actions related to user dataset events.
+ * <p>
+ * UserDatasetEvent Table:
+ * <table>
+ *   <tr>
+ *     <th>Column</th>
+ *     <th>Type</th>
+ *     <th>Notes</th>
+ *   </tr>
+ *   <tr>
+ *     <td><code>EVENT_ID</code></td>
+ *     <td><code>long</code></td>
+ *     <td><code>iRODS generated event ID.</code></td>
+ *   </tr>
+ *   <tr>
+ *     <td><code>USER_DATASET_ID</code></td>
+ *     <td><code>long</code></td>
+ *     <td>ID for the user dataset this event relates to.</td>
+ *   </tr>
+ *   <tr>
+ *     <td><code>HANDLED_TIME</code></td>
+ *     <td>{@link java.time.LocalDateTime}</td>
+ *     <td>Timestamp for when the event was first handled.</td>
+ *   </tr>
+ *   <tr>
+ *     <td><code>STATUS</code></td>
+ *     <td>{@link UserDatasetEventStatus}</td>
+ *     <td>Last updated status for the event.</td>
+ *   </tr>
+ *   <tr>
+ *     <td><code>TYPE_NAME</code></td>
+ *     <td>{@link String}</td>
+ *     <td>Name of the user dataset type.</td>
+ *   </tr>
+ *   <tr>
+ *     <td><code>TYPE_VERSION</code></td>
+ *     <td>{@link String}</td>
+ *     <td>Version of the user dataset type.</td>
+ *   </tr>
+ * </table>
+ */
+public class UserDatasetEventRepo
+{
+  private static final Logger LOG = LogManager.getLogger(UserDatasetEventRepo.class);
+
+  private static final String TABLE_USER_DATASET_EVENT = "userdatasetevent";
+
+  private final String     schema;
+  private final DataSource ds;
+
+  public UserDatasetEventRepo(String schema, DataSource ds) {
+    this.schema = schema;
+    this.ds     = ds;
+  }
+
+  /**
+   * Retrieves the event ID for the last "completed" event.  If no such row is
+   * found, {@code 0} is returned.
+   *
+   * @return The event ID for the last "completed" event if such an event
+   * exists, else returns {@code 0};
+   */
+  public long getLastHandledEvent() {
+    var sql = "SELECT MAX(event_id) FROM " + schema + TABLE_USER_DATASET_EVENT
+      + " WHERE status = '" + UserDatasetEventStatus.COMPLETE + "'";
+
+    var ret = new SQLRunner(ds, sql).executeQuery(rs -> rs.next() ? rs.getLong(1) : 0);
+
+    LOG.debug("Last completed event ID = {}", ret);
+
+    return ret;
+  }
+
+  /**
+   * Updates the status of an event row in the DB to reflect the status value
+   * of the given {@link EventRow} instance.
+   *
+   * @param row Row instance representing the DB row to update.
+   */
+  public void updateEventStatus(EventRow row) {
+    var sql = "UPDATE " + schema + TABLE_USER_DATASET_EVENT
+      + " SET status = ?"
+      + " WHERE event_id = ?";
+
+    new SQLRunner(ds, sql).executeUpdate(
+      new Object[]{row.getStatus().internalValue(), row.getEventID()},
+      new Integer[]{Types.VARCHAR, Types.BIGINT}
+    );
+  }
+
+  /**
+   * Fetches a set of event IDs for events that are in the status
+   * "cleanup_complete".
+   *
+   * @return Set of event IDs.
+   */
+  public Set<Long> getRecoveredEventIDs() {
```

Comment:
why not call this getCleanupCompleteEventIDs()?

Reply:
Renamed